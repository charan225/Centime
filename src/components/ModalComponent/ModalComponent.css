.display-block {
  display: block !important;
}

.modal-backdrop {
  background-color: #fff !important;
  opacity: 0.6 !important;
  z-index: auto;
}

.form-group .required:after {
  content: "*";
  color: red;
}

.new-flow-form-container label {
  float: left;
}
