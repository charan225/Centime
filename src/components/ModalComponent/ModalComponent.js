import React, { useState } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { useTranslation } from "react-i18next";
import { Modal, Button } from "react-bootstrap";

import { rowsActions } from "../../store/index";
import { modalActions } from "../../store/index";

import "./ModalComponent.css";

export const ModalComponent = (props) => {
  const [row, setRow] = useState({});
  const { t } = useTranslation();

  const submitForm = () => {
    props.rowsActions.addRow(row);
    props.modalActions.closeModal();
  };

  const closeModal = () => {
    props.modalActions.closeModal();
  };

  const onChange = (e) => {
    const value = e.target.value;
    const name = e.target.name;
    let newRow = { ...row, [name]: value };
    setRow(newRow);
  };

  const validateForm = () => {
    if (row["inflow"] && row["outflow"] && row["value"]) {
      return false;
    }
    return true;
  };

  return (
    <Modal show={true} onHide={closeModal} animation={false}>
      <Modal.Header closeButton>
        <Modal.Title>{t("Modal_header_Title")}</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <div>
          <form className="new-flow-form-container">
            <div className="form-group">
              <label className="required" htmlFor="inflow-input">
                {t("Inflow_Text")}
              </label>
              <input
                id="inflow-input"
                type="text"
                className="form-control"
                name="inflow"
                onChange={onChange}
              />
            </div>
            <div className="form-group">
              <label className="required" htmlFor="outflow-input">
                {t("Outflow_Text")}
              </label>
              <input
                id="outflow-input"
                type="text"
                className="form-control"
                name="outflow"
                onChange={onChange}
              />
            </div>
            <div className="form-group">
              <label className="required" htmlFor="value-input">
                {t("Value_Text")}
              </label>
              <input
                type="number"
                className="form-control"
                name="value"
                onChange={onChange}
                id="value-input"
              />
            </div>
          </form>
        </div>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" onClick={closeModal}>
          {t("Close_ButtonText")}
        </Button>
        <Button
          variant="primary"
          onClick={submitForm}
          disabled={validateForm()}
        >
          {t("Save_Changes")}
        </Button>
      </Modal.Footer>
    </Modal>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    rowsActions: bindActionCreators(rowsActions, dispatch),
    modalActions: bindActionCreators(modalActions, dispatch),
  };
};

const mapStateToProps = (state) => {
  return {
    isModalOpen: state.modal.showModal,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ModalComponent);
