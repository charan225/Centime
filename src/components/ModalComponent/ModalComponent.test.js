import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { ModalComponent } from "./ModalComponent";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

let modalActions = {
  closeModal: jest.fn(),
  displayModal: jest.fn(),
};

let rowsActions = {
  addRow: jest.fn(),
  upDateRow: jest.fn(),
  deleteRow: jest.fn(),
};

let wrapper, renderComponent;

beforeAll(() => {
  let showModal = false;
  wrapper = shallow(
    <ModalComponent
      rowsActions={rowsActions}
      modalActions={modalActions}
      isModalOpen={showModal}
    />
  );
  renderComponent = (
    <ModalComponent
      rowsActions={rowsActions}
      modalActions={modalActions}
      isModalOpen={showModal}
    />
  );
});

describe("Modal component", () => {
  test("should render Modal content properly", () => {
    const wrapper = shallow(<ModalComponent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("close modal on clicking cancel", () => {
    render(renderComponent);
    const saveButtonElement = screen.getByText("Close_ButtonText");
    userEvent.click(saveButtonElement);
    expect(modalActions.closeModal).toHaveBeenCalled();
  });
});
