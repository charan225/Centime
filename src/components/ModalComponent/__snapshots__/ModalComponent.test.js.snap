// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render Modal content properly 1`] = `
<Modal
  animation={false}
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Modal_header_Title
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <div>
      <form
        className="new-flow-form-container"
      >
        <div
          className="form-group"
        >
          <label
            className="required"
          >
            Inflow_Text
          </label>
          <input
            className="form-control"
            name="inflow"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="required"
          >
            Outflow_Text
          </label>
          <input
            className="form-control"
            name="outflow"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="required"
          >
            Value_Text
          </label>
          <input
            className="form-control"
            name="value"
            onChange={[Function]}
            type="number"
          />
        </div>
      </form>
    </div>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="secondary"
    >
      Close_ButtonText
    </Button>
    <Button
      active={false}
      disabled={true}
      onClick={[Function]}
      variant="primary"
    >
      Save_Changes
    </Button>
  </ModalFooter>
</Modal>
`;
